Add eHealth Knowledge Discovery dataset

mariagrandury opened the pull request:
This Spanish dataset can be used to mine knowledge from unstructured health texts. 

In particular, for:
- Entity recognition
- Relation extraction

lewtun commented on datasets/ehealth_kd/ehealth_kd.py:
```diff
@@ -0,0 +1,159 @@
+_ANNOTATIONS_FILE = "scenario.ann"
+
+
+class eHealthKD(datasets.GeneratorBasedBuilder):
```
I believe that `CamelCase` is the convention for `datasets` so we should probably capitalise the first letter as well, i.e. `EHealthKD(...)`

mariagrandury replied:
You're right, thanks. Done!

lewtun commented on datasets/ehealth_kd/ehealth_kd.py:
```diff
@@ -0,0 +1,159 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
```
I would fill in these TODOs before merging the PR

mariagrandury replied:
Sure, done!

lewtun commented on datasets/ehealth_kd/ehealth_kd.py:
```diff
@@ -0,0 +1,159 @@
+
+    def _generate_examples(self, text_dir, ann_dir):
+        """ Yields examples. """
+        with open(text_dir, encoding="utf-8") as t, open(ann_dir, encoding="utf-8") as a:
```
It would be nice to rename `t` as e.g. `txt_file` and `a` as `ann_file` so the the code downstream is a bit easier to read

mariagrandury replied:
Nice suggestion, done!

lewtun commented on datasets/ehealth_kd/ehealth_kd.py:
```diff
@@ -0,0 +1,159 @@
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls_to_download = {
```
if you wanted, you could save some duplicate code with something like

```python
urls_to_download = {k:[f"{_URL}{v}{_TEXT_FILE}", f"{_URL}{v}{_ANNOTATIONS_FILE}"] for k,v in zip(["train", "dev", "test"], [_TRAIN_DIR, _DEV_DIR, _TEST_DIR])}
```

mariagrandury replied:
Looks really nice, thanks!

lewtun reviewed (changes requested):
Overall it looks really nice! I would suggest filling out the TODOs and add the dataset card (assuming this is still part of the workflow)

mariagrandury commented:
Thank you very much for your review @lewtun ! 

I've updated the script metadata, created the README and fixed the two details you commented.

Ready for another review! 🤗

lewtun reviewed (approved):
LGTM!

mariagrandury commented:
I've updated the task tag as we discussed and also added a couple of lines of code to make sure I include all the available examples.

Thank you again!

lhoestq commented on datasets/ehealth_kd/ehealth_kd.py:
```diff
@@ -0,0 +1,186 @@
+_ANNOTATIONS_FILE = "scenario.ann"
+
+
+class EHealthKD(datasets.GeneratorBasedBuilder):
```
```suggestion
class EhealthKD(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `ehealth_kd`

lhoestq commented on datasets/ehealth_kd/dataset_infos.json:
```diff
@@ -0,0 +1 @@
+{"ehealth_kd": {"description": "Dataset of the eHealth Knowledge Discovery Challenge at IberLEF 2020. It is designed for\nthe identification of semantic entities and relations in Spanish health documents.\n", "citation": "@inproceedings{overview_ehealthkd2020,\n  author    = {Piad{-}Morffis, Alejandro and\n               Guti{'{e}}rrez, Yoan and\n               Ca\u00f1izares-Diaz, Hian and\n     ...
```
```suggestion
{"ehealth_kd": {"description": "Dataset of the eHealth Knowledge Discovery Challenge at IberLEF 2020. It is designed for\nthe identification of semantic entities and relations in Spanish health documents.\n", "citation": "@inproceedings{overview_ehealthkd2020,\n  author    = {Piad{-}Morffis, Alejandro and\n               Guti{'{e}}rrez, Yoan and\n               Ca\u00f1izares-Diaz, Hian and\n               Estevez{-}Velarde, Suilan and\n               Almeida{-}Cruz, Yudivi{'{a}}n and\n               Mu\u00f1oz, Rafael and\n               Montoyo, Andr{'{e}}s},\n  title     = {Overview of the eHealth Knowledge Discovery Challenge at IberLEF 2020},\n  booktitle = ,\n  year      = {2020},\n}\n", "homepage": "https://knowledge-learning.github.io/ehealthkd-2020/", "license": "https://creativecommons.org/licenses/by-nc-sa/4.0/", "features": {"sentence": {"dtype": "string", "id": null, "_type": "Value"}, "entities": [{"ent_id": {"dtype": "string", "id": null, "_type": "Value"}, "ent_text": {"dtype": "string", "id": null, "_type": "Value"}, "ent_label": {"num_classes": 4, "names": ["Concept", "Action", "Predicate", "Reference"], "names_file": null, "id": null, "_type": "ClassLabel"}, "start_character": {"dtype": "int32", "id": null, "_type": "Value"}, "end_character": {"dtype": "int32", "id": null, "_type": "Value"}}], "relations": [{"rel_id": {"dtype": "string", "id": null, "_type": "Value"}, "rel_label": {"num_classes": 13, "names": ["is-a", "same-as", "has-property", "part-of", "causes", "entails", "in-time", "in-place", "in-context", "subject", "target", "domain", "arg"], "names_file": null, "id": null, "_type": "ClassLabel"}, "arg1": {"dtype": "string", "id": null, "_type": "Value"}, "arg2": {"dtype": "string", "id": null, "_type": "Value"}}]}, "post_processed": null, "supervised_keys": null, "builder_name": "ehealth_kd", "config_name": "ehealth_kd", "version": {"version_str": "1.1.0", "description": null, "major": 1, "minor": 1, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 425713, "num_examples": 800, "dataset_name": "ehealth_kd"}, "validation": {"name": "validation", "num_bytes": 108154, "num_examples": 199, "dataset_name": "ehealth_kd"}, "test": {"name": "test", "num_bytes": 47314, "num_examples": 100, "dataset_name": "ehealth_kd"}}, "download_checksums": {"https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/training/scenario.txt": {"num_bytes": 72905, "checksum": "247d41d7c5152d5afb3670e55ccf632d7665f772f42fbd95331b8e65efadaa4e"}, "https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/training/scenario.ann": {"num_bytes": 343367, "checksum": "b4e26cd473cf54bc7e4ad2d5b98896dbeb9b7f4bb5adc426ee2014ce4fce0b88"}, "https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/development/main/scenario.txt": {"num_bytes": 19060, "checksum": "184b5e9a9e69512d5332c81f22d8765ae1e26632e0f5dc089af6e101c9b04149"}, "https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/development/main/scenario.ann": {"num_bytes": 85446, "checksum": "9a47927d13260a10e067d82ebca59d2a43982c7338babb01004c02329611dfb3"}, "https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/testing/scenario3-taskB/scenario.txt": {"num_bytes": 8685, "checksum": "63b6e7ff05445b1fde9c8d9b3bb346a1d9e037858550b4d509fb10d702f682e6"}, "https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/testing/scenario3-taskB/scenario.ann": {"num_bytes": 36437, "checksum": "37102084c1bde2b5eaebc55361b4df7fd0f012495b56f664aa0ad52292a38f00"}}, "download_size": 565900, "post_processing_size": null, "dataset_size": 581181, "size_in_bytes": 1147081}}
```
`e_health_kd`-> `ehealth_kd`

lhoestq reviewed (approved):
Nice thank you !

Really good dataset card as well
